File: front-end/Mobile/src/models/identity/PasswordPolicy.js
import BaseModel from '../BaseModel';

export default class PasswordPolicy extends BaseModel {
  constructor(data = {}) {
    super(data);
    this.policyId = data.policy_id || null;
    this.minLength = data.min_length || 8;
    this.maxLength = data.max_length || 20;
    this.requiresUppercase = data.requires_uppercase !== undefined ? data.requires_uppercase : true;
    this.requiresNumbers = data.requires_numbers !== undefined ? data.requires_numbers : true;
    this.requiresSymbols = data.requires_symbols !== undefined ? data.requires_symbols : true;
    this.expirationDays = data.expiration_days || 90;
  }

  static fromApi(data) {
    if (!data) return null;
    return new PasswordPolicy(data);
  }

  toApi() {
    return {
      ...super.toApi(),
      policy_id: this.policyId,
      min_length: this.minLength,
      max_length: this.maxLength,
      requires_uppercase: this.requiresUppercase,
      requires_numbers: this.requiresNumbers,
      requires_symbols: this.requiresSymbols,
      expiration_days: this.expirationDays,
    };
  }
}

File: back-end/01-ms-identity/src/main/java/com/faceattend_edu/identity_service/domain/model/PasswordPolicy.java
package com.faceattend_edu.identity_service.domain.model;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.Objects;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class PasswordPolicy {
    private Integer policyId;
    private Integer minLength;
    private Integer maxLength;
    private Boolean requiresUppercase;
    private Boolean requiresNumbers;
    private Boolean requiresSymbols;
    private Integer expirationDays;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public void validate() {
        Objects.requireNonNull(this, "PasswordPolicy must not be null");
        if (this.minLength == null || this.minLength < 1) throw new IllegalArgumentException("PasswordPolicy.minLength is required");
        if (this.maxLength == null || this.maxLength < this.minLength) throw new IllegalArgumentException("PasswordPolicy.maxLength must be >= minLength");
        if (this.expirationDays == null || this.expirationDays < 1) throw new IllegalArgumentException("PasswordPolicy.expirationDays is required");
    }
}
